NPE and 404 error when having a @RestQuery field of type Period

### Describe the bug

I have a GET /api REST API that allows to provide a query parameter which is of type 'java.time.Period'

```
    @Schema(description = "Filter by GDPR retention period", nullable = true)
    @RestQuery
    private Period gdprRetentionPeriod;
```

When I call http://localhost:8080/api?gdprRetentionPeriod=P10Y, I get an 404 Not Found Error and a NPE in the logs with a  very cryptic log message : 

**_Caused by: java.lang.NullPointerException: Cannot invoke "org.jboss.resteasy.reactive.server.core.parameters.converters.ParameterConverter.convert(Object)" because "this.quarkusConverter" is null_**

> 19:08:19 DEBUG [WebApplicationException] (executor-thread-2) Application failed the request: jakarta.ws.rs.NotFoundException: HTTP 404 Not Found
> 	at com.nexpublica.bd.ged.document_type.dto.DocumentTypeFiltersDTO.__quarkus_rest_inject(DocumentTypeFiltersDTO.java)
> 	at org.jboss.resteasy.reactive.server.core.parameters.InjectParamExtractor.extractParameter(InjectParamExtractor.java:26)
> 	at org.jboss.resteasy.reactive.server.handlers.ParameterHandler.handle(ParameterHandler.java:45)
> 	at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:179)
> 	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
> 	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$15.runWith(VertxCoreRecorder.java:645)
> 	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
> 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
> 	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
> 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
> 	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
> 	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
> 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
> 	at java.base/java.lang.Thread.run(Thread.java:1583)
> Caused by: java.lang.NullPointerException: Cannot invoke "org.jboss.resteasy.reactive.server.core.parameters.converters.ParameterConverter.convert(Object)" because "this.quarkusConverter" is null
> 	at org.jboss.resteasy.reactive.server.core.parameters.converters.RuntimeResolvedConverter.convert(RuntimeResolvedConverter.java:25)
> 	... 14 more
> 

P10Y is a valid string representation of a Period.

Note that this works perfectly when providing a Period in a JSON body of a  POST query (jackson correctly maps the provided string to a Period).




### Expected behavior

The string representing the Period is correctly converted to the period parameter annotation with @RestQuery

### Actual behavior

Getting a NPE when having a  @RestQuery parameter of type Period.

### How to Reproduce?

Unfortunatly not currently having the time to create a reproducer project.

### Output of `uname -a` or `ver`

Microsoft Windows [Version 10.0.19045.6332]

### Output of `java -version`

temunrin 21.0.8

### Quarkus version or git rev

3.27.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

maven 3.9.9

### Additional information

_No response_

--- a/independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java
+++ b/independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java
@@ -45,6 +45,7 @@ import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNa
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.PATH;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.PATH_PARAM;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.PATH_SEGMENT;
+import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.PERIOD;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.PRIMITIVE_BOOLEAN;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.PRIMITIVE_CHAR;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.PRIMITIVE_DOUBLE;
@@ -159,7 +160,7 @@ public abstract class EndpointIndexer<T extends EndpointIndexer<T, PARAM, METHOD
             RESOURCE_INFO);
 
     protected static final Set<DotName> SUPPORT_TEMPORAL_PARAMS = Set.of(INSTANT, LOCAL_DATE, LOCAL_TIME, LOCAL_DATE_TIME,
-            OFFSET_TIME, OFFSET_DATE_TIME, ZONED_DATE_TIME, YEAR, YEAR_MONTH);
+            OFFSET_TIME, OFFSET_DATE_TIME, ZONED_DATE_TIME, YEAR, YEAR_MONTH, PERIOD);
 
     protected static final Logger log = Logger.getLogger(EndpointIndexer.class);
     protected static final String[] EMPTY_STRING_ARRAY = new String[] {};

--- a/independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/ResteasyReactiveDotNames.java
+++ b/independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/ResteasyReactiveDotNames.java
@@ -11,6 +11,7 @@ import java.time.LocalDateTime;
 import java.time.LocalTime;
 import java.time.OffsetDateTime;
 import java.time.OffsetTime;
+import java.time.Period;
 import java.time.Year;
 import java.time.YearMonth;
 import java.time.ZonedDateTime;
@@ -207,6 +208,7 @@ public final class ResteasyReactiveDotNames {
     public static final DotName ZONED_DATE_TIME = DotName.createSimple(ZonedDateTime.class.getName());
     public static final DotName YEAR = DotName.createSimple(Year.class.getName());
     public static final DotName YEAR_MONTH = DotName.createSimple(YearMonth.class.getName());
+    public static final DotName PERIOD = DotName.createSimple(Period.class.getName());
 
     public static final DotName UNI = DotName.createSimple(Uni.class.getName());
     public static final DotName MULTI = DotName.createSimple(Multi.class.getName());

--- a/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/ServerEndpointIndexer.java
+++ b/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/ServerEndpointIndexer.java
@@ -18,6 +18,7 @@ import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNa
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.MULTI_VALUED_MAP;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.OFFSET_DATE_TIME;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.OFFSET_TIME;
+import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.PERIOD;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.SET;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.SORTED_SET;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.YEAR;
@@ -81,6 +82,7 @@ import org.jboss.resteasy.reactive.server.core.parameters.converters.OffsetTimeP
 import org.jboss.resteasy.reactive.server.core.parameters.converters.OptionalConverter;
 import org.jboss.resteasy.reactive.server.core.parameters.converters.ParameterConverterSupplier;
 import org.jboss.resteasy.reactive.server.core.parameters.converters.PathSegmentParamConverter;
+import org.jboss.resteasy.reactive.server.core.parameters.converters.PeriodParamConverter;
 import org.jboss.resteasy.reactive.server.core.parameters.converters.RuntimeResolvedConverter;
 import org.jboss.resteasy.reactive.server.core.parameters.converters.SetConverter;
 import org.jboss.resteasy.reactive.server.core.parameters.converters.SortedSetConverter;
@@ -636,6 +638,8 @@ public class ServerEndpointIndexer
             return new YearParamConverter.Supplier(format, dateTimeFormatterProviderClassName);
         } else if (YEAR_MONTH.equals(paramType)) {
             return new YearMonthParamConverter.Supplier(format, dateTimeFormatterProviderClassName);
+        } else if (PERIOD.equals(paramType)) {
+            return new PeriodParamConverter.Supplier();
         }
 
         throw new RuntimeException(

--- a/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/PeriodParamConverter.java
+++ b/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/PeriodParamConverter.java
@@ -1,0 +1,24 @@
+package org.jboss.resteasy.reactive.server.core.parameters.converters;
+
+import java.time.Period;
+
+public class PeriodParamConverter implements ParameterConverter {
+
+    @Override
+    public Object convert(Object value) {
+        return Period.parse(value.toString());
+    }
+
+    public static final class Supplier implements ParameterConverterSupplier {
+
+        @Override
+        public ParameterConverter get() {
+            return new PeriodParamConverter();
+        }
+
+        @Override
+        public String getClassName() {
+            return PeriodParamConverter.class.getName();
+        }
+    }
+}

--- a/independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/simple/PeriodParamTest.java
+++ b/independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/simple/PeriodParamTest.java
@@ -1,0 +1,51 @@
+package org.jboss.resteasy.reactive.server.vertx.test.simple;
+
+import static org.hamcrest.Matchers.equalTo;
+
+import java.time.Period;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.QueryParam;
+
+import org.jboss.resteasy.reactive.server.vertx.test.framework.ResteasyReactiveUnitTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.restassured.RestAssured;
+
+public class PeriodParamTest {
+
+    @RegisterExtension
+    static ResteasyReactiveUnitTest test = new ResteasyReactiveUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(PeriodResource.class));
+
+    @Test
+    public void periodAsQueryParam() {
+        RestAssured.get("/period?value=P1Y2M3D")
+                .then().statusCode(200).body(equalTo("P1Y2M3D"));
+    }
+
+    @Test
+    public void periodAsPathParam() {
+        RestAssured.get("/period/P2Y")
+                .then().statusCode(200).body(equalTo("P2Y"));
+    }
+
+    @Path("period")
+    public static class PeriodResource {
+
+        @GET
+        public String query(@QueryParam("value") Period period) {
+            return period.toString();
+        }
+
+        @GET
+        @Path("{value}")
+        public String path(@PathParam("value") Period period) {
+            return period.toString();
+        }
+    }
+}

